src/check-compatibility.ts

```diff
@@ -0,0 +1,67 @@
+import { Connection, Client } from "@temporalio/client";
+import { Worker } from "@temporalio/worker";
+import {
+  getConnectionOptions,
+  getWorkflowOptions,
+  namespace,
+  taskQueue,
+} from "./env";
+
+async function run() {
+  const prTitle = process.argv[2];
+  if (!prTitle) {
+    throw new Error(`usage: ${process.argv[1]} <prTitle>`);
+  }
+  const matches = prTitle.match(/^\[compatible-([^\]]+)\]/);
+  if (matches == null) {
+    throw new Error(`invalid PR title ${prTitle}`);
+  }
+  const buildId = matches[1];
+
+  const connection = await Connection.connect(await getConnectionOptions());
+  const client = new Client({
+    connection,
+    namespace,
+  });
+
+  const compatibility =
+    await client.taskQueue.getBuildIdCompatability(taskQueue);
+  const set = compatibility?.versionSets.find((s) =>
+    s.buildIds.includes(buildId)
+  );
+  if (set == null) {
+    throw new Error(`Could not find set that contains build ID: ${buildId}`);
+  }
+  const joinedBuildIds = set.buildIds.map((b) => `"versioned-${b}"`).join(", ");
+
+  // Based on your use case, you may want to construct a series of more complex queries to validate workflows from
+  // different time ranges and different builds to diversify the coverage.
+  const histories = client.workflow
+    .list({
+      query: `TaskQueue="${taskQueue}" AND BuildIds IN (${joinedBuildIds})`,
+    })
+    .intoHistories();
+  const replayResults = Worker.runReplayHistories(
+    getWorkflowOptions(),
+    histories
+  );
+  let i = 0;
+  for await (const result of replayResults) {
```

Comment:
Not that it's hugely important, but it's maybe slightly performative to do this loop after we actually ran the histories.

Maybe actually important, the limiter doesn't really limit anything meaningful.

It'd be best to make a basic wrapper for the async iterator that does this so it actually happens during `runReplayHistories`

Reply:
Oh wait nevermind I forgot the output is also an async iterator